frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java

```diff
@@ -0,0 +1,69 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IWithLogger;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.gitmachete.frontend.ui.providerservice.BranchLayoutWriterProvider;
+import com.virtuslab.gitmachete.frontend.ui.providerservice.GraphTableProvider;
+import com.virtuslab.gitmachete.frontend.ui.providerservice.SelectedGitRepositoryProvider;
+
+public abstract class BaseProjectDependentAction extends DumbAwareAction implements IWithLogger {
+  @Override
+  @UIEffect
+  public final void update(AnActionEvent anActionEvent) {
+    super.update(anActionEvent);
+
+    var maybeProject = anActionEvent.getProject();
+    var presentation = anActionEvent.getPresentation();
+    if (maybeProject == null) {
+      presentation.setEnabledAndVisible(false);
+    } else {
+      presentation.setEnabledAndVisible(true);
+      onUpdate(anActionEvent);
```

Comment:
Also! To prevent actions from becoming stateful (i.e. so that: what happens in `update` invocation #N should NOT depend on what happened in invocation #N-1), I've ensured that `update` resets the action state. It's enabled & visible by default if project key available, otherwise disabled & hidden; the prior state of the action doesn't matter.